src/core/Track.cpp

```diff
@@ -734,6 +735,9 @@ void TrackContentObjectView::paintTextLabel(QString const & text, QPainter & pai
  */
 void TrackContentObjectView::mousePressEvent( QMouseEvent * me )
 {
+	QVector<TrackContentObjectView *> active = getClickedTCOs();
```

Comment:
I wish there were a way to only use this in the `else if` blocks since this adds overhead to the left click event as well.

Reply:
How about this?

`auto active = me->button() == Qt::RightButton ? getClickedTCOs(): QVector();`

Reply:
I'm not sure how this could be done without repeating code, because both `Qt::LeftButton` and `Qt::MidButton` use the `active` QVector at the moment.

It could be moved to inside the `if` blocks but we would repeat the `getClickedTCOs()` line 4 times. Or doing something similar to what @Spekular suggested and having `auto active = ( me->button() == Qt::LeftButton || Qt::MidButton ) ? QVector() : getClickedTCOs();`. Personally, between the two I think I prefer having `getClickedTCOs` repeated on each `if` block just before the action call, something like:

```
if( me->modifiers() & Qt::ControlModifier )
{
    active = getClickedTCOs();
    remove( active );
}
```

It doesn't look bad (even though the code is repeated a few times), would be more clear and avoids calling the `QVector()` constructor unnecessarily. Should I do it that way?

Reply:
In case you saw the original comment and missed the edit, note that you don't need to check several modifiers. The options are left/mid/right, and active is used in mid and right, so checking for left is enough.

I also doubt the empty QVector() constructor takes much time at all.

If you *do* want to move the declaration into the if statements, please do it only once for right and once for middle click (i.e. two repetitions). There's no reason to do it once per action when all actions  in the branchrequire it.

Reply:
You're right, I misunderstood the logic when I commented. Your solution is drier and I'm up to using it! The idea of having the `getClickedTCOs` before the action was kind of getting ahead of myself thinking about if later we added actions that didn't use the `active` list but were activated by the Right/Mid buttons. Probably not worth it to get those extra lines thinking of something that might not even get added.

Reply:
I'm cool with Spekular's way. 👍